x/stakeibc/keeper/msg_server_claim_undelegated_tokens.go

```diff
@@ -62,15 +87,20 @@ func (k msgServer) ClaimUndelegatedTokens(goCtx context.Context, msg *types.MsgC
 	}
 	timeout := uint64(epoch_tracker.NextEpochStartTime) + uint64(k.GetParam(ctx, types.KeyICATimeoutNanos))
 
-	sequence, err := k.SubmitTxs(ctx, hostZone.GetConnectionId(), msgs, *redemptionAccount, timeout)
-	if err != nil {
-		k.Logger(ctx).Error(fmt.Sprintf("Submit tx error: %s", err.Error()))
-		return nil, err
+	icaTx := IcaTx{
+		ConnectionId: hostZone.GetConnectionId(),
+		Msgs:         msgs,
+		Account:      *redemptionAccount,
+		Timeout:      timeout,
 	}
 
+	return &icaTx, nil
+}
+
+func (k Keeper) FlagRedemptionRecordsAsClaimed(ctx sdk.Context, userRedemptionRecord *recordstypes.UserRedemptionRecord, sequence uint64) {
 	// Set isClaimable to false, so that the record can't be claimed again
 	userRedemptionRecord.IsClaimable = false
-	k.RecordsKeeper.SetUserRedemptionRecord(ctx, userRedemptionRecord)
+	k.RecordsKeeper.SetUserRedemptionRecord(ctx, *userRedemptionRecord)
```

Comment:
Why this change out of curiosity?

Reply:
Since the main function was split up, the child functions needed a userRedmptionRecord and I could have either passed the object itself, or a reference. I went with a reference since this seemed more in line with other types of this flavor.
```
# before
func ClaimUndelegatedTokens() {
    userRedemptionRecord := ... // object
    k.RecordsKeeper.SetUserRedemptionRecord(ctx, userRedemptionRecord)
}

# after
func ClaimUndelegatedTokens() {
    userRedemptionRecordPointer := k.GetClaimableRedemptionRecord(ctx, msg) // returns pointer
    k.FlagRedemptionRecordsAsClaimed(ctx, userRedemptionRecordPointer, sequence)
}
func FlagRedemptionRecordsAsClaimed(userRedemptionRecordPointer) {
   k.RecordsKeeper.SetUserRedemptionRecord(ctx, *userRedemptionRecordPointer)
}
```